cli_tools/gce_onestep_image_import/onestep_importer/aws_importer.go

```diff
@@ -0,0 +1,346 @@
+)
+
+// awsImporter is responsible for importing image from AWS
+type awsImporter struct {
+	args           *awsImportArguments
+	gcsClient      domain.StorageClientInterface
+	ctx            context.Context
+	oauth          string
+	paramPopulator param.Populator
+
+	// AWS clients for SDK
+	ec2Client *ec2.EC2
+	s3Client  *s3.S3
+}
+
+// newAWSImporter creates an new awsImporter instance.
+// Automatically populating dependencies, such as compute/storage clients.
+func newAWSImporter(oauth string, args *awsImportArguments) (*awsImporter, error) {
+	ctx := context.Background()
+	client, err := createGCSClient(ctx, oauth)
+	if err != nil {
+		return nil, err
+	}
+
+	computeClient, err := param.CreateComputeClient(&ctx, oauth, args.gcsComputeEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	metadataGCE := &compute.MetadataGCE{}
+	paramPopulator := param.NewPopulator(
+		metadataGCE,
+		client,
+		storageutils.NewResourceLocationRetriever(metadataGCE, computeClient),
+		storageutils.NewScratchBucketCreator(ctx, client),
+	)
+
+	awsSession, err := createAWSSession(args.region, args.accessKeyID, args.secretAccessKey, args.sessionToken)
+	if err != nil {
+		return nil, err
+	}
+
+	importer := &awsImporter{
+		args:           args,
+		gcsClient:      client,
+		s3Client:       s3.New(awsSession),
+		ec2Client:      ec2.New(awsSession),
+		ctx:            ctx,
+		oauth:          oauth,
+		paramPopulator: paramPopulator,
+	}
+
+	return importer, nil
+}
+
+// createGCSClient creates a new GCS client.
+func createGCSClient(ctx context.Context, oauth string) (domain.StorageClientInterface, error) {
+	log.SetPrefix(logPrefix + " ")
+	logger := logging.NewStdoutLogger(logPrefix)
+
+	baseTransport := &http.Transport{
+		DisableKeepAlives:     false,
+		MaxIdleConns:          0,
+		MaxIdleConnsPerHost:   1000,
+		MaxConnsPerHost:       0,
+		IdleConnTimeout:       60 * time.Second,
+		ResponseHeaderTimeout: 5 * time.Second,
+		TLSHandshakeTimeout:   5 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+	}
+	transport, err := htransport.NewTransport(ctx, baseTransport)
+	if err != nil {
+		return nil, err
+	}
+
+	return storageutils.NewStorageClient(ctx, logger, option.WithHTTPClient(&http.Client{Transport: transport}),
+		option.WithCredentialsFile(oauth))
+}
+
+// createAWSSession creates a new Session for AWS SDK.
+func createAWSSession(region, accessKeyID, secretAccessKey, sessionToken string) (*session.Session, error) {
+	return session.NewSession(&aws.Config{
+		Region: aws.String(region),
+		Credentials: credentials.NewStaticCredentials(
+			accessKeyID,
+			secretAccessKey,
+			sessionToken),
+	})
+}
+
+// Run runs aws import workflow.
+func (importer *awsImporter) run() (string, error) {
+	//0. validate AWS args
+	err := importer.args.validateAndPopulate(importer.paramPopulator)
+	if err != nil {
+		return "", err
+	}
+
+	// 1. export AMI to AWS S3 if user did not specify an exported AMI path.
+	if importer.args.exportedAMIPath == "" {
```

Comment:
I saw a couiple of places used importer.args.exportedAMIPath == "". It's better to make a IsExportRequired() bool